File: algorithm/src/main/java/baekjoon/p03062/Main.java
package baekjoon.p03062;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            System.out.println(solve(sc.nextInt()));
        }
    }

    private static String solve(int input) {
        if (sym(input + rev(input))) {
            return "YES";
        } else {
            return "NO";
        }
    }

    private static int rev(int n) {
        String str = String.valueOf(n);
        String rev = new StringBuilder(str).reverse().toString();

        return Integer.parseInt(rev);
    }

    private static boolean sym(int n) {
        String str = String.valueOf(n);

        for (int i = 0; i < str.length(); i++) {
            if (str.charAt(i) != str.charAt(str.length() - i - 1)) {
                return false;
            }
        }

        return true;
    }

}

File: algorithm/src/main/java/baekjoon/p03062/Main.kt
package baekjoon.p03062

fun main(args: Array<String>) {
    repeat(readLine()!!.toInt()) {
        println(solve(readLine()!!.toInt()))
    }
}

private fun solve(input: Int): String {
    val sum = input + input.toString().reversed().toInt()
    val str = sum.toString()

    return when (str == str.reversed()) {
        true -> "YES"
        false -> "NO"
    }
}